### ISSUE TITLE
Workspace dependency management

### ISSUE THREAD

**Issue by ggwpez** [paritytech/substrate#13859]
Try out what Kilt did and use workspace dependencies to ease the version managment.  
Example [Cargo.toml](https://github.com/KILTprotocol/kilt-node/blob/eeb90d1045dc4157b3713beb8e6c7667154fe166/runtimes/common/Cargo.toml#L41) and [Workspace](https://github.com/KILTprotocol/kilt-node/blob/eeb90d1045dc4157b3713beb8e6c7667154fe166/Cargo.toml#L25).  

quote @bkchr 
>I would say we start with some like log etc. Deps that are shared between most crates and see how it goes. 

Any known downsides with this @weichweich?

**Comment by weichweich** [comment]
@ggwpez It's nice to have everything in one place. Especially with none substrate/cumulus/polkadot dependencies. We for example had a different version for the same dependency in all crates. This doesn't happen anymore.

The only downside I noticed is that the feature management is somewhat unclear. You now have to check two places to be sure about which features are enabled for a crate.
I also think the intuition might be that the features specified on the workspace level are overwritten if features are specified in the crate, but the [docs state that features are additive](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#inheriting-a-dependency-from-a-workspace). So features enabled in the workspace are always on.

**Comment by ggwpez** [comment]
Thanks for the reply!  
Not having multiple versions is probably the biggest improvement, from a management and compile time perspective.  
Will take a look at the other things.

We probably want to do this after the mono-repo migration, I assume.

**Comment by gilescope** [comment]
Yeah no reason to slow down the mono-repo switch to wait for this and I think it will be nicer to do once we're all in one workspace. Jan said there were a couple of crates that absolutely had to be the same version or things didn't work (wasmtime might have been one?).
